src/com/generic/util/Persistent.java

```diff
@@ -57,5 +59,40 @@ private void parseWarehouseContentsToObjects(JSONObject shipmentObject) {
 		// add the shipment to the warehouse
 		warehouseTracker.addShipment(warehouseID, shipment);
 	}
-
+	
+	public void parseXML(String filepath) throws IOException, ParseException {
+		File inputFile = new File(filepath);
+		WarehouseTracker warehouseTracker = WarehouseTracker.getInstance();
+		DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+		DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+		Document doc = dBuilder.parse(inputFile);
+		doc.getDocumentElement().normalize();
+		NodeList nList = doc.getElementsByTagName("Warehouse");
+		for (int i = 0; i < nList.getLength(); i++) {
+			Node nNode = nList.item(i);
+			if (nNode.getNodeType() == Node.ELEMENT_NODE) {
+				Element eWarehouseElement = (Element) nNode;
+				String warehouseID = eWarehouseElement.getAttribute("id");
```

Comment:
Remember that this is 2nd iteration, requirements have changed, now the warehouse has a name attribute.